src/server.js

```diff
@@ -80,7 +83,7 @@ app.use((req, res) => {
     return;
   }
 
-  match({ history, routes: getRoutes(store), location: req.originalUrl }, (error, redirectLocation, renderProps) => {
+  match({ syncedHistory, routes: getRoutes(store), location: req.originalUrl }, (error, redirectLocation, renderProps) => {
```

Comment:
hi, in official example, for server side used syncedHistory 
https://github.com/reactjs/react-router-redux/blob/v4.0.0/examples/server/server.js#L43